datafusion/core/src/sql/planner.rs

```diff
@@ -97,9 +97,10 @@ fn plan_key(key: SQLExpr) -> Result<ScalarValue> {
             ScalarValue::Int64(Some(s.parse().unwrap()))
         }
         SQLExpr::Value(Value::SingleQuotedString(s)) => ScalarValue::Utf8(Some(s)),
+        SQLExpr::Identifier(Ident { value, .. }) => ScalarValue::Utf8(Some(value)),
```

Comment:
Looks likes, there are mistakes in tests, I found that current tests are not valid:

In select.rs, you need to do some changes:

```
// " - is used for identifiers, but in this test, should be some_struct['bar'], because it's a name of the field, not an identifier.
let sql = "SELECT some_struct[\"bar\"] as l0 FROM structs LIMIT 3";
```

Reply:
Shall we do this as part of a follow on PR? Or would you like to do it as part of this one?

Reply:
Probably, it's a good idea to drop this line and change tests to correct variant: `some_struct['bar']`. Right now, GetIndexedExpr doesn't support dynamic expression as `key`, and converting `Identifier` to `ScalarValue::Utf8` is not correct, because it can be a variable for example.

Reply:
will do

Reply:
In 09d972221c